lib/virtio_mmio/virtio_mmio.c

```diff
@@ -29,6 +29,26 @@ extern void virtio_device_reset(struct virtio_device *vdev);
 
 void virtio_mmio_isr(struct virtio_device *vdev);
 
+#if defined(HVL_VIRTIO)
+int virtio_mmio_hvl_wait_cfg(struct virtio_device *vdev, int offset, uint32_t data);
+
+/* weak symbols */
+void virtio_mmio_shm_pool_init(void *mem, size_t size) __attribute__((weak));
```

Comment:
Using weak function in library is quite dangerous, we faced issue when we tried to use "weak" function in  libmetal. So I would prefer to avoid such mechanism

Reply:
Please share some information on the type of issues you encountered. In this case I think weak symbols provide a more readable implementation. Replacing them with function pointers is the other available option.